File: PDA.BACK/Microservices/Services/Store/GkdEntryService.cs
using HWMX.DotNet;
using HWMX.DotNet.ORM; 
using Microservices.Interfaces.Store;
using Microsoft.AspNetCore.Http;
using Repositories.Database.Store;
using Repositories.Interfaces.Store; 

namespace Microservices.Services.Store
{
    public class GkdEntryService(IGkdEntryRepository _repository, IHttpContextAccessor _httpContext) : IGkdEntryService {

        public async Task<ResponseList<LOT_GKD_ENTRY_DTO>> GetKDLotInfo(string lotNumber)
        {
            ResponseList<LOT_GKD_ENTRY_DTO> response = new();

            try
            {
                ResponseProcedure responseProcedure = await _repository.GetKDLotInfo(lotNumber);

                if (responseProcedure.Failure)
                    return response.Error(responseProcedure.MessageList); 

                response.Data = responseProcedure.GetTable<LOT_GKD_ENTRY_DTO>();
            }

            catch (Exception ex)
            {
                return response.Exception(ex);
            }

            return response;
        }
         

        public async Task<ResponseDTO<string>> SetKdStockIn(List<LOT_GKD_ENTRY_DTO> lotList)
        {
            ResponseDTO<string> response = new();
            try
            {
                string LotNumber      = string.Join(";", lotList.Select(x => x.LOT_NUMBER));
                string PartNumber     = string.Join(";", lotList.Select(x => x.PART_NUMBER));
                string Qty            = string.Join(";", lotList.Select(x => x.QTY));
                string EoNumber       = string.Join(";", lotList.Select(x => x.EO_NUMBER));
                string Unit           = string.Join(";", lotList.Select(x => x.UNIT));
                string VendorCode     = string.Join(";", lotList.Select(x => x.VENDOR_CODE));
                string ProductionDate = string.Join(";", lotList.Select(x => x.PRODUCTION_DATE));

                HttpRequestDTO httpContext = _httpContext.ToHttpRequest();
                ResponseProcedure responseProcedure = await _repository.SetKdStockIn(LotNumber, PartNumber, Qty, Unit, ProductionDate, EoNumber, VendorCode, httpContext.User);

                if (responseProcedure.Failure)
                    return response.Error(responseProcedure.MessageList);

                response.Data = responseProcedure.GetOutput("IO_MESSAGE");

                if (response.Data != "OK")
                    return response.BadRequest(response.Data);

                response.Data = LANGUAGE.MESSAGE.SuccessfulTransaction(httpContext.Language);
            }

            catch (Exception ex)
            {
                return response.Exception(ex);
            }

            return response;
        }


        public async Task<ResponseDTO<string>> SetKdStockAoneIn(string vbelg)
        {
            ResponseDTO<string> response = new();
            try
            {
                HttpRequestDTO httpContext = _httpContext.ToHttpRequest();
                ResponseProcedure responseProcedure = await _repository.SetKdStockAoneIn(vbelg, httpContext.User);

                if (responseProcedure.Failure)
                    return response.Error(responseProcedure.MessageList);

                response.Data = responseProcedure.GetOutput("IO_MESSAGE");

                if (response.Data != "OK")
                    return response.BadRequest(response.Data);

                response.Data = LANGUAGE.MESSAGE.SuccessfulTransaction(httpContext.Language);
            }

            catch (Exception ex)
            {
                return response.Exception(ex);
            }

            return response;
        }
    }
}

File: PDA.FRONT/src/app/modules/store/gkd-entry/gkd-entry.service.ts
import { Injectable } from "@angular/core"; 
import { appSettings } from "@appSettings"; 
import { HTTP } from "hwmx-angular/tools";  

@Injectable({ providedIn: 'root' })
export class GkdEntryService extends HTTP {

    private readonly controller = `${appSettings.webAPI.hwmxPDA}/api/Store/GkdEntry`;   

    /** GET */
    public GetKDLotInfo = async (barcode: string) =>  {
        const response = await HTTP.GET<GKD_ENTRY_DTO[]>({
            url: `${this.controller}/GetKDLotInfo/${barcode}`
        }); 
        
        if(!response.ok) {  
            if(response.status < 500) {
                this.alert.Warning(response.message, barcode, 'barcode'); 
            } 

            else {  
                this.alert.Danger('GetKDLotInfo', 'Error', 'bug'); 
                console.error(response.message); 
            }
            
            return [];
        }  

        if(response.data.length <= 0)
            this.alert.Warning('No Data', barcode, 'barcode'); 

        return response.data.map(item => ({ ...item, Status: 0 }));
    }  


    /** HTTP POST */
    public SetKdStockIn = async (lotList: any[]) =>  {
        const response = await HTTP.POST<string>({
            url: `${this.controller}/SetKdStockIn`,
            responseType: 'text', 
            body: lotList
        });  

        if(!response.ok) {             
            if(response.status < 500) {
                this.alert.Warning(response.message, null, 'barcode'); 
            }
    
            else {
                this.alert.Danger('SetKdStockIn', 'Error', 'bug'); 
                console.error(response.message);
            }
        } 

        return response.data;
    }


    /** HTTP POST */
    public SetKdStockAoneIn = async (vbelg: string) =>  {
        const response = await HTTP.POST<string>({
            url: `${this.controller}/SetKdStockAoneIn/${vbelg}`,
            responseType: 'text' 
        }); 

        if(!response.ok) {             
            if(response.status < 500) {
                this.alert.Warning(response.message, vbelg, 'barcode'); 
            }
    
            else {
                this.alert.Danger('SetKdStockAoneIn', 'Error', 'bug'); 
                console.error(response.message);
            }
        }  

        return response.data;
    } 
}  


export interface GKD_ENTRY_DTO {
    LOT_NUMBER:      string;
    PART_NUMBER:     string;
    QTY:             number;
    EO_NUMBER:       string;
    UNIT:            string;
    VENDOR_CODE:     string;
    PRODUCTION_DATE: string;
    VBELG:           string;
    STATUS:          number;
} 
